client/ayon_core/hosts/maya/plugins/workfile_build/load_placeholder.py

```diff
@@ -0,0 +1,264 @@
+        selection = cmds.ls(selection=True)
+        if len(selection) > 1:
+            raise ValueError("More then one item are selected")
+
+        parent = selection[0] if selection else None
+
+        placeholder_data["plugin_identifier"] = self.identifier
+
+        placeholder_name = self._create_placeholder_name(placeholder_data)
+
+        placeholder = cmds.spaceLocator(name=placeholder_name)[0]
+        if parent:
+            placeholder = cmds.parent(placeholder, selection[0])[0]
+
+        imprint(placeholder, placeholder_data)
+
+        # Add helper attributes to keep placeholder info
+        cmds.addAttr(
+            placeholder,
+            longName="parent",
+            hidden=True,
+            dataType="string"
+        )
+        cmds.addAttr(
+            placeholder,
+            longName="index",
+            hidden=True,
+            attributeType="short",
+            defaultValue=-1
+        )
+
+        cmds.setAttr(placeholder + ".parent", "", type="string")
+
+    def update_placeholder(self, placeholder_item, placeholder_data):
+        node_name = placeholder_item.scene_identifier
+        new_values = {}
+        for key, value in placeholder_data.items():
+            placeholder_value = placeholder_item.data.get(key)
+            if value != placeholder_value:
+                new_values[key] = value
+                placeholder_item.data[key] = value
+
+        for key in new_values.keys():
+            cmds.deleteAttr(node_name + "." + key)
+
+        imprint(node_name, new_values)
+
+    def collect_placeholders(self):
+        output = []
+        scene_placeholders = self._collect_scene_placeholders()
+        for node_name, placeholder_data in scene_placeholders.items():
+            if placeholder_data.get("plugin_identifier") != self.identifier:
+                continue
+
+            # TODO do data validations and maybe upgrades if they are invalid
+            output.append(
+                LoadPlaceholderItem(node_name, placeholder_data, self)
+            )
+
+        return output
+
+    def populate_placeholder(self, placeholder):
+        self.populate_load_placeholder(placeholder)
+
+    def repopulate_placeholder(self, placeholder):
+        repre_ids = self._get_loaded_repre_ids()
+        self.populate_load_placeholder(placeholder, repre_ids)
+
+    def get_placeholder_options(self, options=None):
+        return self.get_load_plugin_options(options)
+
+    def post_placeholder_process(self, placeholder, failed):
+        """Cleanup placeholder after load of its corresponding representations.
+
+        Args:
+            placeholder (PlaceholderItem): Item which was just used to load
+                representation.
+            failed (bool): Loading of representation failed.
+        """
+        # Hide placeholder and add them to placeholder set
+        node = placeholder.scene_identifier
+
+        cmds.sets(node, addElement=PLACEHOLDER_SET)
```

Comment:
```suggestion
        # If we just populate the placeholders from current scene, the
        # placeholder set will not be created so account for that.
        if not cmds.objExists(PLACEHOLDER_SET):
            cmds.sets(name=PLACEHOLDER_SET, empty=True)

        cmds.sets(node, addElement=PLACEHOLDER_SET)
```

Reply:
This is an existing bug in the Maya placeholder loading, which is not yet fixed.

Reply:
Just to cross-reference the PR: https://github.com/ynput/ayon-core/pull/459

Reply:
Since there is now another PR that does this I'll resolve this conversation and leave it up to that PR to get merged or vice-versa and resolve conflicts when they occur?

Reply:
Yeah, sounds good.